src/main/groovy/com/twcable/gradle/cqpackage/CreatePackageTask.groovy

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twcable.gradle.cqpackage
+
+import groovy.transform.TypeChecked
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.CopySpec
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.bundling.Zip
+
+import javax.annotation.Nonnull
+
+import static com.twcable.gradle.cqpackage.CqPackageUtils.allBundleFiles
+import static com.twcable.gradle.cqpackage.CqPackageUtils.nonProjectDependencyBundleFiles
+import static com.twcable.gradle.cqpackage.CqPackageUtils.projectBundleJarFile
+import static com.twcable.gradle.cqpackage.CreatePackageTask.CopyBundlesMode.ALL
+import static com.twcable.gradle.cqpackage.CreatePackageTask.CopyBundlesMode.NONE
+import static com.twcable.gradle.cqpackage.CreatePackageTask.CopyBundlesMode.NON_PROJECT_ONLY
+import static com.twcable.gradle.cqpackage.CreatePackageTask.CopyBundlesMode.PROJECT_ONLY
+
+@TypeChecked
+class CreatePackageTask extends Zip {
+    private String _bundleInstallRoot = '/apps/install'
+    private File _contentSrc
+
+    /**
+     * Mutable collection of exclusions to apply to copying files from {@link #getContentSrc()}.
+     * Defaults with a "reasonable" set, such as "*&#42;/.vlt", "*&#42;/.git/**", etc.
+     * Generally to modify this list you would mutate this in-place.
+     */
+    Collection<String> fileExclusions
+
+    CopyBundlesMode copyBundlesMode = ALL
+
+    /**
+     * The Configuration to use when determining what bundles to put in the package.
+     * @see #addAllBundles()
+     */
+    Configuration configuration
+
+
+    CreatePackageTask() {
+        super()
+
+        // AbstractTask() guarantees that this.project has been set
+
+        setDefaults()
+        initialize()
+    }
+
+
+    private void setDefaults() {
+        setDescription("Creates the CQ Package zip file")
+        setGroup("CQ")
+
+        // don't know why this isn't being set automatically, but...
+        setBaseName(project.name)
+        setVersion(project.version as String)
+
+        fileExclusions = [
+            '**/.git',
+            '**/.git/**',
+            '**/.gitattributes',
+            '**/.gitignore',
+            '**/.gitmodules',
+            '**/.vlt',
+            'jcr_root/.vlt-sync-config.properties',
+            'jcr_root/var/**',
+            'SLING-INF/**',
```

Comment:
Wouldn't be a good idea to have a configurable file other than these basic exclusions so this list can be extended?